src/account-info/AccountDetailsScreen.js

```diff
@@ -1,38 +1,50 @@
+/* @flow */
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 
+import { Auth } from '../types';
 import boundActions from '../boundActions';
 import { getAuth } from '../account/accountSelectors';
 import { Screen } from '../common';
 import AccountDetails from './AccountDetails';
 import { getCurrentRoute } from '../nav/routingSelectors';
 
+const NOT_FOUND_USER = {
+  fullName: 'User not found',
+  status: '',
+  avatarUrl: ''
+};
+
 class AccountDetailsScreen extends Component {
 
   props: {
-    fullName: string,
+    auth: Auth,
     email: string,
     avatarUrl: string,
-  }
+    users: Object[],
+    fetchMessages: () => void,
+    doNarrow: (string) => void,
+    popRoute: (string) => void
+  };
+  user: Object;
 
   componentWillMount() {
     // props.email gets reset during navigation slide out (on back)
-    // so we cache the value to prevent an exception
-    this.email = this.props.email;
+    // so we cache values to prevent an exception
+    this.user = this.props.users.find(x => x.email === this.props.email) || NOT_FOUND_USER;
```

Comment:
@borisyankov @neerajwahi I refactored as @neerajwahi suggested - take a look at it please